src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftRepository.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.ccd.data.draft;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.ccd.domain.model.draft.CreateCaseDataContentDraft;
+import uk.gov.hmcts.ccd.domain.model.draft.Draft;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ServiceException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+
+@Named
+@Qualifier(DefaultDraftRepository.QUALIFIER)
+@Singleton
+public class DefaultDraftRepository implements DraftRepository {
```

Comment:
Could we rename those to `DraftGateway` and `DefaultDraftGateway` instead to avoid confusion with DAL?
For reference, see:
- https://martinfowler.com/eaaCatalog/gateway.html
- https://martinfowler.com/eaaCatalog/repository.html